alchemy/src/stripe/portal-configuration.ts

```diff
@@ -0,0 +1,544 @@
+import Stripe from "stripe";
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+
+/**
+ * Business profile information for the portal
+ */
+export interface PortalConfigurationBusinessProfile {
+  /**
+   * The messaging shown to customers in the portal
+   */
+  headline?: string;
+  /**
+   * A link to the business's publicly available privacy policy
+   */
+  privacyPolicyUrl?: string;
+  /**
+   * A link to the business's publicly available terms of service
+   */
+  termsOfServiceUrl?: string;
+}
+
+/**
+ * Features available in the customer portal
+ */
+export interface PortalConfigurationFeatures {
+  /**
+   * Information about updating the customer details in the portal
+   */
+  customerUpdate?: {
+    /**
+     * The types of customer update that are supported
+     */
+    allowedUpdates?: Array<Stripe.BillingPortal.ConfigurationCreateParams.Features.CustomerUpdate.AllowedUpdate>;
+    /**
+     * Whether the feature is enabled
+     */
+    enabled?: boolean;
+  };
+  /**
+   * Information about showing the billing history in the portal
+   */
+  invoiceHistory?: {
+    /**
+     * Whether the feature is enabled
+     */
+    enabled?: boolean;
+  };
+  /**
+   * Information about updating payment methods in the portal
+   */
+  paymentMethodUpdate?: {
+    /**
+     * Whether the feature is enabled
+     */
+    enabled?: boolean;
+  };
+  /**
+   * Information about canceling subscriptions in the portal
+   */
+  subscriptionCancel?: {
+    /**
+     * Whether to cancel subscriptions immediately or at the end of the billing period
+     */
+    cancellationReason?: {
+      /**
+       * Whether the feature is enabled
+       */
+      enabled?: boolean;
+      /**
+       * Which cancellation reasons will be given as options to the customer
+       */
+      options?: Array<Stripe.BillingPortal.ConfigurationCreateParams.Features.SubscriptionCancel.CancellationReason.Option>;
+    };
+    /**
+     * Whether the feature is enabled
+     */
+    enabled?: boolean;
+    /**
+     * Whether to cancel subscriptions immediately or at the end of the billing period
+     */
+    mode?: Stripe.BillingPortal.ConfigurationCreateParams.Features.SubscriptionCancel.Mode;
+    /**
+     * Whether to create prorations when canceling subscriptions
+     */
```

Comment:
ideally nested object typers are given theuir own interface

ideally we would use the stripe types where possible too (wuthout going too crazy with things like Omit)